src/metatrain/utils/omegaconf.py

```diff
@@ -126,6 +126,112 @@ def _resolve_single_str(config: str) -> DictConfig:
 CONF_ENERGY["stress"] = CONF_GRADIENT.copy()
 
 
+def check_unknown_options(target: DictConfig, reference: DictConfig):
```

Comment:
I know that this function is quite nested. I am thinking about a refactoring. I already have some ideas but probably this makes sense when trying to check the dataset options.